llvm/utils/TableGen/README.md

```diff
@@ -0,0 +1,54 @@
+# LLVM TableGen
+
+The purpose of TableGen is to generate complex output files based on information
+from source files that are significantly easier to code than the output files would be, and also easier to maintain and modify over time.
+
+The information is coded in a declarative style involving classes and records,
+which are then processed by TableGen.
+
+```
+class Hello <string _msg> {
+	string msg = !strconcat("Hello ", _msg);
```

Comment:
nit: why is the indentation here different from that of line 19 and 23?

Reply:
I wrote it in Compiler Explorer where <tab> is an actual tab, llvm-tblgen outputs 2 space tabs I presume. I've fixed it.